include/openssl/asn1.h

```diff
@@ -276,7 +276,8 @@ typedef struct ASN1_VALUE_st ASN1_VALUE;
 # define TYPEDEF_I2D_OF(type) typedef int i2d_of_##type(const type *,unsigned char **)
 # define TYPEDEF_D2I2D_OF(type) TYPEDEF_D2I_OF(type); TYPEDEF_I2D_OF(type)
 
-TYPEDEF_D2I2D_OF(void);
+typedef void *d2i_of_void(void **, const unsigned char **, long);
+typedef int i2d_of_void(const void *, unsigned char **);
```

Comment:
Since you restored `TYPEDEF_D2I2D_OF`, the remaining change in this file is rather pointless, wouldn't you say?

Reply:
I agree it's not necessary, but it's far from pointless.

The macros have one public use and that's not necessary.  This makes the transition of eventually removing them easier.  And also, the code is much more clear now IMO.

Reply:
Okie, I can accept that.